### PR TITLE
NPM: Allow multiple lockfiles to be present

### PR DESCRIPTION
**Pull request by sschuberth**
[1] says that "The file created and updated by this command will then take
precedence over any other existing or future package-lock.json files",
so there is no conflict between "npm-shrinkwrap.json" and
"package-lock.json" files. Similarly, there is no conflict between
"yarn.lock" and "package-lock.json" files as the former is only used by
Yarn, and we will use Yarn if (and only if) "yarn.lock" is present.

While at it, slightly improve the wording of error and debug messages.

[1] https://docs.npmjs.com/cli/shrinkwrap

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/595)

### CODE REVIEW

File: analyzer/src/funTest/kotlin/NpmTest.kt [hunk 1]
```diff
@@ -114,24 +114,6 @@ class NpmTest : WordSpec() {
                 yamlMapper.writeValueAsString(result) shouldBe expectedResult
             }
 
-            "show error if multiple lockfiles are present" {
```
Comment (mnonnenmacher) [inline analyzer/src/funTest/kotlin/NpmTest.kt]: This test should be replaced with one that verifies the new behaviour, currently "if NPM and Yarn lockfiles are present Yarn is used". But also see my other comment.

**Comment by mnonnenmacher** [comment]
If both "package-lock.json" and "yarn.lock" are present preferring Yarn is an arbitrary choice. Shouldn't we instead resolve both? They could generate different results and there should be a reason that both lock files exist. For this the "provider" should be set to "Yarn" for Yarn packages, which we should change regardless of the outcome of the question before.

**Comment by sschuberth** [comment]
> If both "package-lock.json" and "yarn.lock" are present preferring Yarn is an arbitrary choice.

You have good points there. Still I wonder whether we should spend time on implementing this rather edgy case, or we simply leave the behavior as it currently is.

File: analyzer/src/main/kotlin/managers/NPM.kt [hunk 2]
```diff
@@ -481,12 +470,14 @@ class NPM : PackageManager() {
                 packages.map { it.toCuratedPackage() }.toSortedSet())
     }
 
+    protected open fun getLockFiles() = listOf("npm-shrinkwrap.json", "package-lock.json")
```
Comment (mnonnenmacher) [inline analyzer/src/main/kotlin/managers/NPM.kt]: This should be an `open val`, not a function:
https://kotlinlang.org/docs/reference/classes.html#overriding-properties

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/NPM.kt]: Done.

**Pull request merged**